client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -878,7 +881,19 @@ function mediaButton( editor ) {
 	editor.on( 'touchstart touchmove touchend', selectImageOnTap() );
 
 	editor.on( 'init', function () {
-		MediaStore.on( 'change', updateMedia );
+		unsubscribeFromReduxStore = store.subscribe( () => {
+			const siteId = getSelectedSiteId( store.getState() );
+			if ( ! siteId ) {
+				return;
+			}
+
+			previousMediaReference = currentMediaReference;
+			currentMediaReference = getMedia( store.getState(), siteId );
+
+			if ( previousMediaReference !== currentMediaReference ) {
```

Comment:
This line depends on #44986 being merged first, otherwise it'd be rather inefficient.